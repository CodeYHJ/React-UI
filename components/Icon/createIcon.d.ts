export interface CreateIconProps {
    scriptUrl: string;
}
declare const createIcon: (scriptUrl: string) => import("react").SFC<import("./localIcon").IconProps>;
export default createIcon;
