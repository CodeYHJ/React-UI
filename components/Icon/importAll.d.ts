declare let importAll: (requireContext: any) => any;
