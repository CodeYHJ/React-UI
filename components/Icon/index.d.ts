import I from './localIcon';
import c from './createIcon';
declare type IconType = typeof I & {
    createIcon: typeof c;
};
declare const Icon: IconType;
export default Icon;
