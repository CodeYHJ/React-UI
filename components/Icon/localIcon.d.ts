import React, { SVGAttributes } from 'react';
import "./importAll";
import "./style/index.less";
export interface IconProps extends SVGAttributes<SVGElement> {
    name: string;
}
declare const Icon: React.SFC<IconProps>;
export default Icon;
