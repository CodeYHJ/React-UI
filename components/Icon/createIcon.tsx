import Icon from './localIcon';

export interface CreateIconProps {
  scriptUrl: string;
}

const setMap = new Set();

const createIcon = (scriptUrl: string) => {
  if (!setMap.has(scriptUrl)) {
    const svgDom = document.createElement('script');

    svgDom.setAttribute('src', scriptUrl);

    document.body.appendChild(svgDom);

    setMap.add(scriptUrl);
  }

  return Icon;
};

export default createIcon;
