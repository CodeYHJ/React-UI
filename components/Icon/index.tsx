import I from './localIcon';

import c from './createIcon';

export * from './svgIcon';

type IconType = typeof I & { createIcon: typeof c };

const Icon = I as IconType;

Icon.createIcon = c;

export default Icon;
