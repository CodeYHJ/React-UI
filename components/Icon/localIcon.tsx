import React, { SVGAttributes } from 'react';

import './style/index.less';

import { classPre } from '@com/utils';

export interface IconProps extends SVGAttributes<SVGElement> {
  name: string;
}

const c = classPre('svg');

const Icon: React.FunctionComponent<IconProps> = (props) => {
  const { className, name, ...others } = props;

  const cls = [c(), className].filter(Boolean).join(' ');

  return (
    <svg className={cls} {...others}>
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

export default Icon;
