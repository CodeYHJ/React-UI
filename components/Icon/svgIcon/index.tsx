export { default as ArrorDownIcon } from './ArrorDownIcon';

export { default as ArrorLeftIcon } from './ArrorLeftIcon';

export { default as ArrorRightIcon } from './ArrorRightIcon';

export { default as ArrorUpIcon } from './ArrorUpIcon';

export { default as CancelIcon } from './CancelIcon';

export { default as CodeIcon } from './CodeIcon';

export { default as DangerIcon } from './DangerIcon';

export { default as LoadingIcon } from './LoadingIcon';

export { default as MainIcon } from './MainIcon';

export { default as PasswordIcon } from './PasswordIcon';

export { default as SuccessIcon } from './SuccessIcon';

export { default as UserIcon } from './UserIcon';

export { default as WarnIcon } from './WarnIcon';
