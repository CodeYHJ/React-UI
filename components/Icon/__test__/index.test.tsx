import Icon from "../index"
const { createIcon, MainIcon } = Icon
import React from "react"
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
describe('localIcon', () => {
    it('renders an `.code-ui-svg` component', () => {
        const json = renderer.create(<MainIcon />)
        expect(json).toMatchSnapshot()
    })
    it('trigger onOK once when click on svg', () => {
        const onClick = jest.fn()
        const wrapper = shallow(<MainIcon onClick={onClick} />);
        wrapper.find('.code-ui-svg').simulate('click')
        expect(onClick).toBeCalled()
    })
})

describe('createIcon', () => {
    it('return a localIcon', () => {
        const LocalIcon = createIcon('//at.alicdn.com/t/font_1973431_ia5jq3fwut.js')
        expect(Icon).toEqual(LocalIcon)
    })

})