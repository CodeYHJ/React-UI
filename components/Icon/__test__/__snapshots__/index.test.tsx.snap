// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localIcon renders an \`.code-ui-svg\` component 1`] = `
<svg
  className="code-ui-svg"
  viewBox="0 0 1024 1024"
>
  <path
    d="M553.984 553.984v-256h-84.01v256h84.01zm0 172.032V640h-84.01v86.016h84.01zM512 86.016q176 0 301.013 125.013t125.014 301.014-125.014 301.013T512 938.069 210.987 813.056 85.973 512.043t125.014-301.014T512 86.016z"
  />
</svg>
`;
